vignettes/secuTrialR-package-vignette.Rmd

```diff
@@ -1,172 +1,570 @@
+Thus, a whole study may have 99% completeness on variable basis while showing
+0% completeness on form basis. It is currently not technically possible to assess
+completeness on variable basis in a generic way. Hence, high completeness on form
+basis implies high completeness on variable basis but **NOT** vice versa.
+
+For a more participant id centred statistic you can perform the following.
+\vspace{5pt}
 
 ```{r}
-head(sT_export_shortnames$bmd[,c("pat_id", "age", "grouping", "bmd")], n = 4)
+fsc <- form_status_counts(ctu05_data)
+# show the top
+head(fsc)
+```
+
+## Form linkage
 
-all.equal(sT_export_shortnames$bmd, sT_export_longtnames$dem00bmd)
+Linkages amongst forms can be explored with the `links_secuTrial()` function.
+This relies on the `igraph` package to create a network. It is possible to
+interact with the network, e.g. move nodes around in order to read the labels better.
+The device ID is returned to the console, but can be ignored. Forms are plotted in
+deep yellow, variables in light blue.
+\vspace{5pt}
 
+```{r, eval = FALSE}
+links_secuTrial(ctu05_data)
 ```
 
+The output can not be shown within this vignette but you should give it a try.
+Please note that the linkage plot is likely most useful **without** the audit trail
+data in the export.
+
+## Sampling random participants
+
+During study monitoring it is common practice to check random participants
+from a study database. These participants should be retrieved in a reproducible
+fashion. The below function allows this for a loaded secuTrial data export.
+\vspace{5pt}
+
+```{r}
+# retrieve at least 25 percent of participants recorded after March 18th 2019 
+# from the centres "Inselspital Bern" and "Charité Berlin"
+return_random_participants(ctu05_data,
+                           percent = 0.25,
+                           seed = 1337,
+                           date = "2019-03-18",
+                           centres = c("Inselspital Bern (RPACK)",
+                                       "Charité Berlin (RPACK)"))
+```
 
-### Analysing data completeness
+Please note that earlier R versions may return different results because
+there is a different `rng_config`.
+For this reason we have added the `rng_config` to the output.
 
-For various reasons eCRFs are frequently only partially filled. This can be commonly
-observed in cohort studies of registries. Thus, before performing more thorough analyses
-it is sometimes informative to get an overview of data completeness.
+## Retrieve score variables
 
-To asses data completeness of a secuTrial export you first need to generate the
-validation overview in secuTrial and export it as an *.xlsx file. Remember to select the
-"Column" and "Completion status" columns before you export the validation overview.
-The validation overview for this vignette is also part of this package.
+Generally, it is advisable to recalculate score variables before data analysis.
```

Comment:
I didn't know about "score variables" before. For me, the information provided here does not suffice to comprehend what the score variables are.

Reply:
This is for English (its different in German):
If you grep for `"calculated only"` in `items$itemtype` you can find the items which are scores. i.e. scores are secuTrial data types. 
I will add a brief paragraph explaining scores a little closer.